lib/registrant.rb

```diff
@@ -0,0 +1,45 @@
+class Registrant
+    
+    attr_reader :name,
+                :age,
+                :permit,
+                :license_data,
+                :take_written_test
+
+    def initialize(name, age, permit = false)
+        @name = name
+        @age = age
+        @permit = permit
+        @license_data = {
+            :written => false,
+            :license => false,
+            :renewed => false
+        }
+    end
+
+    def permit?
+        @permit
+    end
+
+    def earn_permit
+        @permit = true
+    end
+
+    def take_written_test
+        if @license_data[:written] = true
+            true
+        else @age < 16 && permit? == false
+            false
+        end
```

Comment:
I'm not sure why we have all these conditions in this method. We already tested all these conditions in the Facility class method.